tests/scenarios/protocol-message-size-limits.spec.ts

```diff
@@ -0,0 +1,109 @@
+
+    it('should enforce protocol message size limits', async () => {
+      // Scenario:
+      // Alice joins a size limited protocol
+      // 1. Alice should be able to write a message that does not exceed the protocol message size limit
+      // 2. Alice should not be able to write a message that exceeds the protocol message size limit
+
+      // creating Alice and Bob persona and setting up a stub DID resolver
+      const alice = await TestDataGenerator.generatePersona();
+      TestStubGenerator.stubDidResolver(didResolver, [alice]);
+
+      const protocolDefinition: ProtocolDefinition = dataSizeLimitProtocolDefinition as ProtocolDefinition;
+
+      // Alice configures protocol with encryption
+      const protocolDefinitionForAlice
+        = await Protocols.deriveAndInjectPublicEncryptionKeys(protocolDefinition, alice.keyId, alice.keyPair.privateJwk);
+      const protocolsConfigureForAlice = await TestDataGenerator.generateProtocolsConfigure({
+        author             : alice,
+        protocolDefinition : protocolDefinitionForAlice
+      });
+
+      const protocolsConfigureForAliceReply = await dwn.processMessage(
+        alice.did,
+        protocolsConfigureForAlice.message
+      );
+      expect(protocolsConfigureForAliceReply.status.code).to.equal(202);
+
+      // 1. Alice writes a message to her own DWN within data size limit
+      const blobBytes = TestDataGenerator.randomBytes(1000);
+      const blobRecord = await TestDataGenerator.generateProtocolEncryptedRecordsWrite({
+        plaintextBytes                                   : blobBytes,
+        author                                           : alice,
+        protocolDefinition                               : protocolDefinition,
+        protocolPath                                     : 'blob',
+        encryptSymmetricKeyWithProtocolPathDerivedKey    : false,
+        encryptSymmetricKeyWithProtocolContextDerivedKey : true
+      });
+      const blobRecordReply1 = await dwn.processMessage(alice.did, blobRecord.message, { dataStream: blobRecord.dataStream });
+      expect(blobRecordReply1.status.code).to.equal(202);
+
+      // 2. Alice writes a message to her own DWN exceeding data size limit
+      const blobBytes2 = TestDataGenerator.randomBytes(1001);
+      const blobRecord2 = await TestDataGenerator.generateProtocolEncryptedRecordsWrite({
+        plaintextBytes                                   : blobBytes2,
+        author                                           : alice,
+        protocolDefinition                               : protocolDefinition,
+        protocolPath                                     : 'blob',
+        encryptSymmetricKeyWithProtocolPathDerivedKey    : false,
+        encryptSymmetricKeyWithProtocolContextDerivedKey : true
+      });
+      const blobRecordReply2 = await dwn.processMessage(alice.did, blobRecord2.message, { dataStream: blobRecord2.dataStream });
+      expect(blobRecordReply2.status.code).to.equal(400);
+
```

Comment:
Sanity: check for the specific error code.
```suggestion
      expect(blobRecordReply2.status.code).to.equal(400);
      expect(blobRecordReply2.status.detail).to.contain(DwnErrorCode.ProtocolAuthorizationSizeLimitExceeded);
```